### PR TITLE
Add self-test failed event handling in ota demos

### PR DESCRIPTION
**Pull request by pvyawaha**
*Issue #, if available:*

*Description of changes:*

* Add handling for self-test failed event in OTA over MQTT & HTTP demos

* Remove disconnect handling in HTTP deinit n OTA over HTTP demo

* Minor refactor otaAppCallback by using switch case for OTA over MQTT & HTTP demo

* Add delay  with logs before exiting the application 

* some minor updates on macros

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: demos/ota/ota_demo_core_http/ota_demo_core_http.c [hunk 1]
```diff
@@ -119,7 +119,7 @@
 /**
  * @brief Transport timeout in milliseconds for transport send and receive.
  */
-#define TRANSPORT_SEND_RECV_TIMEOUT_MS           ( 500 )
+#define TRANSPORT_SEND_RECV_TIMEOUT_MS           ( 500u )
```
Comment (yanjos-dev) [inline demos/ota/ota_demo_core_http/ota_demo_core_http.c:122]: Can you make this an uppercase U for consistency?

File: demos/ota/ota_demo_core_http/ota_demo_core_http.c [hunk 2]
```diff
@@ -729,53 +734,73 @@ static void otaAppCallback( OtaJobEvent_t event,
...
+            OTA_Shutdown( 0 );
 
-        /* Requires manual activation of new image.*/
-        LogError( ( "New image activation failed." ) );
-    }
-    else if( event == OtaJobEventFail )
-    {
-        LogInfo( ( "Received OtaJobEventFail callback from OTA Agent." ) );
+            /* Requires manual activation of new image.*/
+            LogError( ( "New image activation failed." ) );
```
Comment (yanjos-dev) [inline demos/ota/ota_demo_core_http/ota_demo_core_http.c:749]: Would it make sense to add to this log that the user should manually activate the image? As a user, I would see this in the console and think something went wrong.

File: demos/ota/ota_demo_core_http/ota_demo_core_http.c (hunk 1 above)
Comment (pvyawaha) [reply demos/ota/ota_demo_core_http/ota_demo_core_http.c]: Fixed

File: demos/ota/ota_demo_core_http/ota_demo_core_http.c (hunk 2 above)
Comment (yanjos-dev) [reply demos/ota/ota_demo_core_http/ota_demo_core_http.c:749]: I see, that makes sense. Thanks.

**Review by yanjos-dev — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Review by yanjos-dev — APPROVED** [review]

**Pull request merged**